### DISCUSSION TITLE
Error displaying widget: model not found in JupyterHub

### DISCUSSION THREAD

**Discussion by tony-maynard** [idom-team/idom#599] [Problem]
I am following the IDOM tutorial and idom-jupyter but experiencing a problem. Steps taken:

- pip Install idom[stable]
- pip install idom_jupyter

Notebook Content:
```py
import idom
import idom_jupyter


@idom.component
def ClickCount():
    count, set_count = idom.hooks.use_state(0)
    return idom.html.h1("Hello, World!")


ClickCount()
```
Output: `Error displaying widget: model not found`
Some package versions:

```
idom                      0.33.3           
idom-jupyter              0.6.2  
jupyterlab-widgets        1.0.2                    
jupyter_core              4.9.1 
jupyterlab                3.3.0a1
```

I'm not sure what's exactly going wrong it looks like all my extension are installed correctly, any help would be appreciated please:
```
jupyterlab-plotly v5.5.0 enabled OK
        idom-client-jupyter v0.4.0 enabled OK (python, idom_jupyter)
        jupyterlab-code-snippets v2.1.0 enabled OK (python, code_snippet)
        jupyterlab-system-monitor v0.8.0 enabled OK (python, jupyterlab-system-monitor)
        jupyterlab-topbar-extension v0.6.1 enabled OK (python, jupyterlab-topbar)
        @jupyter-widgets/jupyterlab-manager v3.0.1 enabled OK (python, jupyterlab_widgets)
        @jupyter-server/resource-usage v0.6.1 enabled OK (python, jupyter-resource-usage)
        @krassowski/jupyterlab-lsp v3.8.1 enabled OK (python, jupyterlab-lsp)

Other labextensions (built into JupyterLab)
   app dir: /opt/conda/envs/XXX/share/jupyter/lab
        @jupyter-widgets/base v4.0.0 enabled OK
```

**Comment by rmorshea** [comment]
I'm unable to reproduce at the moment. Here are some things to try:

- Try it with `jupyter notebook` instead of `jupyter lab` - see if a similar problem arises
- Try installing in a fresh environment (see note below before doing this though).

If the issue persists, I'd be happy to hop on a call to help resolve this. Feel free to grab a slot during [office hours](https://calendly.com/rmorshea/60min?month=2022-01).

At the moment, if you install in a fresh environment, run `pip install idom-jupyter` instead of `pip install idom[stable] idom-jupyter`. The latter seems to install an incompatible version of `idom` and `idom-jupyter`. I suspect that `pip` is backtracking to an earlier versions of `idom-jupyter` which have an unbounded dependency on `idom`.

**Reply by tony-maynard** [reply]
Thanks @rmorshea, apologies I have been away since submitting this issue. I have tried running just  `pip install idom-jupyter` and noticed in the console I am receiving the following error 
<img width="603" alt="Screenshot 2022-02-01 at 09 13 27" src="https://user-images.githubusercontent.com/85884885/151941473-9f3dd838-1d09-4dee-bdd9-c14c193494b2.png">

**Reply by tony-maynard** [reply]
It turns out that I had to install idom_jupyter in the base conda env and the one I was using, do you know why that was the case?

**Reply by rmorshea** [reply]
I found a couple of similar issues:

- https://github.com/matplotlib/ipympl/issues/161
- https://github.com/jupyter-widgets/widget-ts-cookiecutter/issues/39

But it's hard to say whether your issue is an example of the ones above.

**Reply by rmorshea** [reply]
I think I figured out what's going on here. I forgot to bump the version on various things in `idom-jupyter`. Not sure why it was working at all TBH :shrug:

**Reply by Archmonger** [reply]
We definitely should think of a way to automate version bumps 😅

**Reply by rmorshea** [reply]
The `idom-jupyter` repo is a bit tough in this regard. The instructions that came with the repo template I used also don't talk about all the versiony things you need to manage either so that didn't help.

**Reply by rmorshea** [reply]
Here's all the spots with versions.

- https://github.com/idom-team/idom-jupyter/blob/e3a9fc0ecffd9b01a0e4be7e678ac99d10e56213/idom_jupyter/_version.py#L2
- https://github.com/idom-team/idom-jupyter/blob/e3a9fc0ecffd9b01a0e4be7e678ac99d10e56213/idom_jupyter/widget.py#L58-L61
- https://github.com/idom-team/idom-jupyter/blob/e3a9fc0ecffd9b01a0e4be7e678ac99d10e56213/js/package.json#L3
- https://github.com/idom-team/idom-jupyter/blob/e3a9fc0ecffd9b01a0e4be7e678ac99d10e56213/js/lib/widget.js#L11-L12

I suppose all these versions give you a way to finely communication where changes happened, but I think I'm just gonna try and keep them in sync.

**Reply by tony-maynard** [reply]
I'm no expert but I suspect this is the case

**Reply by rmorshea** [reply]
the latest release of idom-jupyter should resolve this. it should also allow idom to work in VSCode.

**Comment by tony-maynard** [comment]
Hi @rmorshea I am getting the same error again, `Error: Module idom-client-jupyter, semver range ^0.7.0 is not registered as a widget module` anything I can do to help?

**Reply by tony-maynard** [reply]
Using
```
idom                      0.36.3                   pypi_0    pypi
idom-jupyter              0.7.5

**Reply by rmorshea** [reply]
Yup. Same issue as before. I remembered to do it on the server-side model, but not the client. I'll try to get to this shortly. Till then, you'll need to use idom<0.36.3 with idom-jupyter<0.7.4

**Reply by rmorshea** [reply]
I'll try and automate the version sync in the next release so this stops happening.

**Reply by tony-maynard** [reply]
I see, thanks @rmorshea let me know if I can get involved in some way.

**Reply by rmorshea** [reply]
@tony-maynard if you'd be interested, I could write up some instructions for bumping versions of `idom` in `idom-jupyter`. If you're able to make PRs for those sorts of regular bumps, that would reduce that burden on myself. When breaking changes happen, I'd also be happy to meet up and walk you through how to integrate them with `idom-jupyter` so you'll be able to contribute to those as well.
